rfc/rfc017-distributed-network-grpc-v2.md

```diff
@@ -45,4 +45,20 @@ As specified by RFC015, the node MUST authenticate the peer's node DID as follow
 1. Resolve peer's node DID to its corresponding DID document.
 2. Assert that one of the `NutsComm` endpoint's host matches (one of) the `dNSName` SANs in the peer's TLS client certificate.
 
-## X. Private Transactions
\ No newline at end of file
+## X. Private Transactions
+
+When the node receives a transaction that contains a `pal` header, the transaction is considered private.
+This means the transaction contents can only be retrieved by (and should only be shared with) the participants listed in the `pal` header.
+Since the `pal` header is encrypted (see [RFC004](rfc004-verifiable-transactional-graph.md)) to preserve anonymity of the participants,
+it must be decrypted first using the node's `keyAgreement` keys from its DID document.
+If the local node can decrypt the `pal` header, it means the transaction is (also) intended for the local node.
+See [RFC004](rfc004-verifiable-transactional-graph.md) for more information on how to encrypt/decrypt the `pal` header.
+
+To retrieve the contents of the private transaction, the node MUST send a `TransactionPayloadQuery` to one (or more) of the nodes listed in the `pal` header.
+It COULD decide to broadcast the message to all nodes in the `pal` header (except the local node), because not all nodes (in the `pal` header) might have the contents yet (or never will).
+
+When a `TransactionPayloadQuery` for a private transaction is received, the node MUST decrypt its `pal` header and verify that the requesting peer is listed as participant.
+
+Both `TransactionPayloadQuery` and its success response (`TransactionPayload` with the transaction contents) MUST only be sent over an authenticated connection.
```

Comment:
Should we specify that responses to _not having the payload_ and _requestor is not on pal list_ be the same?
If these are not the same it is possible to determine if a node is on the participants list based on its answer to a `TransactionPayloadQuery`.

Reply:
I intended it to be, but I'll make it explicit